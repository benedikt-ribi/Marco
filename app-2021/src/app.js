import { crud } from "./crud.js";

export class application {
  constructor(rootUrl){
    this.#addresses = new crud(`${rootUrl}addresses`)
  }

  initialize = async () => {
    const listItems = document.getElementById("listItems");

    try {
      listItems.innerHTML = "";
      for (const address of await this.#addresses.getAll()) {
        const link = document.createElement("a");
        link.href = "#";
        const application = this;
        link.addEventListener("click", () => application.#showDetail(address));

        const div = document.createElement("div");
        div.appendChild(link);

        listItems.appendChild(div);
      }
    } catch (e) {
      listItems.innerHTML = e.message;
    }
  };

  /**
   * @param {address} address
   */
  #showDetail(address) {
    const detail = document.getElementById("detail");

    detail.innerHTML = `${address.firstName} ${address.lastName}`;
  }

  /**
   * @type {crud<address>}
   */
  #addresses;
}
