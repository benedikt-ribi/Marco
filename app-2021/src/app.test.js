/**
 * @jest-environment jsdom
 */

import { describe, expect, test } from "@jest/globals";
import { application } from "./app";
import "isomorphic-fetch";

describe("Application", () => {
  test("initialize", async () => {
    document.body.innerHTML = `<div id="listItems"></div>`;

    const listItems = document.getElementById("listItems");

    const app = new application();

    await app.initialize(listItems);

    expect(listItems.innerHTML).toMatchSnapshot();
  });

  test("clicking an item loads its details", async () => {
    document.body.innerHTML = `
      <div id="listItems"></div>
      <div id="detail"></div>
    `;

    const app = new application();
    const listItems = document.getElementById("listItems");
    const detail = document.getElementById("detail");

    await app.initialize(listItems);

    const [firstLink] = listItems.getElementsByTagName("a");

    firstLink.click();

    expect(detail.innerHTML).toMatchSnapshot();
  });
});
