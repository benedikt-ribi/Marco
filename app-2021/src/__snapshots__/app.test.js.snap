// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Application clicking an item loads its details 1`] = `"Peter Smith"`;

exports[`Application initialize 1`] = `"Peter Smith<br>Marco von Ballmoos<br>Hanseeee Maier<br>Kerim Schmitz<br>Peter Schmitz<br>Marco von Ballmoos<br>Hans Maier<br>Kerim Schmitz<br>Peter Schmitz<br>Marco von Ballmoos<br>Hans Maier<br>bob jones<br>"`;
